### PR TITLE
Number lookup ga

### PR DESCRIPTION
**Pull request by ericasp16**
# Description
Update azure-communication-phonenumbers SDK to support GA for number lookup feature

Swagger updates: https://github.com/Azure/azure-rest-api-specs/pull/26294

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.
  - This will be coming in a subsequent version, this is just for review, the service is not deployed with these changes so tests will not succeed currently

### CODE REVIEW

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py [hunk 1]
```diff
@@ -423,3 +430,28 @@ def list_available_area_codes(
...
         )
+
+    @distributed_trace
+    def search_operator_information(
+            self,
+            phone_numbers,  # type: PhoneNumberSearchType
+            options=None, #type: OperatorInformationRequestOptions
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> OperatorInformationResult
```
Comment (kashifkhan) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py:441]: for newer additions we should be using the python 3 typing conventions, so we can remove this line and type the function.

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py [hunk 2]
```diff
@@ -423,3 +430,28 @@ def list_available_area_codes(
                 "administrative_division", None),
             **kwargs
         )
+
+    @distributed_trace
+    def search_operator_information(
+            self,
+            phone_numbers,  # type: PhoneNumberSearchType
+            options=None, #type: OperatorInformationRequestOptions
```
Comment (kashifkhan) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py:438]:
```suggestion
            options: Optional[OperatorInformationRequestOptions]=None,
```

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py [hunk 3]
```diff
@@ -423,3 +430,28 @@ def list_available_area_codes(
...
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> OperatorInformationResult
+        """Searches for operator information for a given list of phone numbers.
+
+        :param phone_numbers: The phone number(s) whose operator information should be searched
+        :type phone_numbers: str or list(str)
```
Comment (kashifkhan) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py:445]:
```suggestion
        :type phone_numbers: str or list[str]
```

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/aio/_phone_numbers_client_async.py [hunk 4]
```diff
@@ -421,6 +427,28 @@ def list_available_area_codes(
             **kwargs
         )
 
+    @distributed_trace
+    def search_operator_information(
+            self,
+            phone_numbers,  # type: PhoneNumberSearchType
```
Comment (kashifkhan) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/aio/_phone_numbers_client_async.py:433]: same changes here to add in py3 typing conventions :)

**Review by xiangyan99 — COMMENTED** [review]
The PR is too big to review.

Can you split it into small ones?

**Pull request merged**